projects/wpdx/src/app/rehab-prio/adm-popup/adm-popup.component.ts

```diff
@@ -0,0 +1,193 @@
+      }
+      if (value.NAME_1) {
+        queries.push(`${baseQuery}
+          where install_year is not NULL and clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' order by install_year`);
+      }
+      if (value.NAME_2) {
+        queries.push(`${baseQuery}
+          where install_year is not NULL and clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
+                clean_adm2='${this.fq(value.NAME_2)}' order by install_year`);
+      }
+      if (value.NAME_3) {
+        queries.push(`${baseQuery}
+          where install_year is not NULL and clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
+                clean_adm2='${this.fq(value.NAME_2)}' and clean_adm3='${this.fq(value.NAME_3)}' order by install_year`);
+      }
+
+      if (value.NAME_4) {
+        queries.push(`${baseQuery}
+          where install_year is not NULL and clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
+                clean_adm2='${this.fq(value.NAME_2)}' and clean_adm3='${this.fq(value.NAME_3)}' and clean_adm4='${this.fq(value.NAME_4)}' order by install_year`);
+      }
+      
+    return queries;                                                                                                                               
+  }
+
+  getDataQuery(value, attribute){
+    const baseQuery = `select ${attribute}, count(${attribute}) from 
+    wpdx_plus`
+    const queries: string[] = [];
+    if (value.NAME_0) {
+      queries.push(`${baseQuery}
+        where clean_country_name='${this.fq(value.NAME_0)}' group by ${attribute}`);
+    }
+    if (value.NAME_1) {
+      queries.push(`${baseQuery}
+        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' group by ${attribute}`);
+    }
+    if (value.NAME_2) {
+      queries.push(`${baseQuery}
+        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
+              clean_adm2='${this.fq(value.NAME_2)}' group by ${attribute}`);
+    }
+    if (value.NAME_3) {
+      queries.push(`${baseQuery}
+        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
+              clean_adm2='${this.fq(value.NAME_2)}' and clean_adm3='${this.fq(value.NAME_3)}' group by ${attribute}`);
+    }
+
+    if (value.NAME_4) {
+      queries.push(`${baseQuery}
+        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
+              clean_adm2='${this.fq(value.NAME_2)}' and clean_adm3='${this.fq(value.NAME_3)}' and clean_adm4='${this.fq(value.NAME_4)}' group by ${attribute}`);
+    }
+    return queries;
+
+  }
+
+  getManagementData(){
+    forkJoin(this.getDataQuery(this.popupProperties, 'management_clean').map(q => this.db.query(q))).subscribe(results => {
```

Comment:
This can also be parametrized and generalized and then used in these three functions (instead of using nearly duplicate code)